cmd/kube-scheduler/app/server.go

```diff
@@ -339,3 +285,70 @@ func WithPlugin(name string, factory framework.PluginFactory) Option {
 		return registry.Register(name, factory)
 	}
 }
+
+// Setup creates a completed config and a scheduler based on the command args and options
+func Setup(ctx context.Context, args []string, opts *options.Options, outOfTreeRegistryOptions ...Option) (*schedulerserverconfig.CompletedConfig, *scheduler.Scheduler, error) {
+	if len(args) != 0 {
+		fmt.Fprint(os.Stderr, "arguments are not supported\n")
+	}
+
+	if errs := opts.Validate(); len(errs) > 0 {
+		return nil, nil, utilerrors.NewAggregate(errs)
+	}
+
+	if len(opts.WriteConfigTo) > 0 {
+		c := &schedulerserverconfig.Config{}
+		if err := opts.ApplyTo(c); err != nil {
+			return nil, nil, err
+		}
+		if err := options.WriteConfigFile(opts.WriteConfigTo, &c.ComponentConfig); err != nil {
```

Comment:
This specific block could you to `Run`. And maybe we shouldn't be calling `ApplyTo`, but just let the Setup go through all the other steps (like `opts.Config`). `write-to-config` is not intended for production, just for our own debugging. So the closer it is to a real run, the better.

Reply:
Do u mean move this block to Run? Maybe this flag is also useful to debug config problem in production env. Moving this to Run will make some differences, for example, the scheduler instance will be created, let me check if there are other unexpected effects.

Reply:
Specially when used in production, you want to see the configuration that is actually going to run. I guess the only effects we want to avoid is that other files get created or routines are started.

Reply:
moved to `Run`